src/main/java/io/fabric8/maven/docker/util/aws/AwsSdkAuthConfigFactory.java

```diff
@@ -0,0 +1,41 @@
+package io.fabric8.maven.docker.util.aws;
+
+import com.amazonaws.SdkClientException;
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.AWSSessionCredentials;
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
```

Comment:
Actually I would go full in with reflection: 

* Avoid any direct import or references to AWS SDK classes
* Check with `Class.forName()` whether a certain class is avalialble. 
* Instantiate object and execute method only via `javax.reflection` means, with proper error handling or fallback (e.g. necessary when there might be different signatures for different versions of the SDK).  For now, I would focus on a single version of the SDK and document that.

Reply:
Got it. Let's see when I find the time for this. :wink:

Reply:
@rhuss got it. Only thing missing is documentation and referring to it in the log.

Can the documentation be safely included in this branch or will it go live before releasing?

Reply:
Documentation is always based on the latest tag. You can just add it to the PR and we can merge on master. During release it will be merged into the dmp.fabric8.io branch from where the documentation is generated by circleci.

Thanks for the update ! I'm just on my way to a long weekend, but will have a look as soon as I'm back.